frame/lending/src/lib.rs

```diff
@@ -846,41 +857,98 @@ pub mod pallet {
+				if let Some(deposit) = BorrowRent::<T>::get(market_id, account) {
+					let market_account = Self::account_id(market_id);
+					<T as Config>::NativeCurrency::transfer(
+						&market_account,
+						liquidator,
+						deposit,
+						false,
+					)?;
+				}
+			} else {
+				return Err(DispatchError::Other(
+					"Tried liquidate position which is not supposed to be liquidated",
+				))
+			}
+			Ok(())
+		}
+
+		/// Liquidates position for each borrower in the vector within mentioned market.
+		/// Returns a vector of borrowers' account ids whose positions were requested to
+		/// be liquidated.
+		/// - `liquidator` : Liquidator's account id.
+		/// - `market_id` : Market index from which `borrowers` has taken borrow.
+		/// - `borrowers` : Vector of borrowers whose debts are going to be liquidated.
 		pub fn liquidate_internal(
 			liquidator: &<Self as DeFiEngine>::AccountId,
 			market_id: &<Self as Lending>::MarketId,
-			borrowers: Vec<<Self as DeFiEngine>::AccountId>,
-		) -> Result<(), DispatchError> {
+			borrowers: BoundedVec<<Self as DeFiEngine>::AccountId, T::MaxLiquidationBatchSize>,
+		) -> Result<Vec<<Self as DeFiEngine>::AccountId>, DispatchError> {
+			// Vector of borrowers whose positions are involved in the liquidation process.
+			let mut subjected_borrowers: Vec<<Self as DeFiEngine>::AccountId> = Vec::new();
+			let market_pair = Self::get_market(market_id)?;
+			let borrow_asset = T::Vault::asset_id(&market_pair.1.borrow_asset_vault)?;
 			for account in borrowers.iter() {
-				if Self::should_liquidate(market_id, account)? {
-					let market = Self::get_market(market_id)?;
-					let borrow_asset = T::Vault::asset_id(&market.borrow_asset_vault)?;
-					let collateral_to_liquidate = Self::collateral_of_account(market_id, account)?;
-					let source_target_account = Self::account_id(market_id);
-					let unit_price = T::Oracle::get_ratio(CurrencyPair::new(
-						market.collateral_asset,
-						borrow_asset,
-					))?;
-					let sell = Sell::new(
-						market.collateral_asset,
-						borrow_asset,
-						collateral_to_liquidate,
-						unit_price,
-					);
-					T::Liquidation::liquidate(&source_target_account, sell, market.liquidators)?;
-
-					if let Some(deposit) = BorrowRent::<T>::get(market_id, account) {
-						let market_account = Self::account_id(market_id);
-						<T as Config>::NativeCurrency::transfer(
-							&market_account,
-							liquidator,
-							deposit,
-							false,
-						)?;
+				// Wrap liquidate position request in a storage transaction.
+				// So, in the case of any error state's changes will not be committed
+				let storage_transaction_succeeded = with_transaction(|| {
+					let liquidation_response_result =
+						Self::liquidate_position(liquidator, &market_pair, borrow_asset, account);
+					if let Err(error) = liquidation_response_result {
+						log::error!("Creation of liquidation request for position {:?} {:?} was failed: {:?}",
```

Comment:
may be return error out of tx and log warning, not error. 
error usually handled layer +1 out of scope and errors will allow spam

Reply:
Anyway we return Result here. Log level was changed.